src/main/kotlin/uk/gov/gdx/datashare/service/EventPollService.kt

```diff
@@ -1,78 +1,55 @@
 
+import kotlinx.coroutines.FlowPreview
 import kotlinx.coroutines.flow.*
 import org.slf4j.Logger
 import org.slf4j.LoggerFactory
 import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
 import uk.gov.gdx.datashare.config.AuthenticationFacade
-import uk.gov.gdx.datashare.repository.DataConsumerRepository
+import uk.gov.gdx.datashare.repository.ConsumerSubscriptionRepository
 import uk.gov.gdx.datashare.repository.EventDataRepository
-import uk.gov.gdx.datashare.resource.EventType
 import uk.gov.gdx.datashare.resource.SubscribedEvent
 import java.time.LocalDateTime
 
 @Service
 class EventPollService(
   private val eventDataRepository: EventDataRepository,
   private val authenticationFacade: AuthenticationFacade,
-  private val dataConsumerRepository: DataConsumerRepository,
+  private val consumerSubscriptionRepository: ConsumerSubscriptionRepository
 ) {
   companion object {
     val log: Logger = LoggerFactory.getLogger(this::class.java)
   }
 
+  @OptIn(FlowPreview::class)
+  @Transactional
   suspend fun getEvents(
-    eventTypes: List<EventType>?,
+    eventTypes: List<String>?,
     fromTime: LocalDateTime?,
     toTime: LocalDateTime?
   ): Flow<SubscribedEvent> {
-    val oauthClient = authenticationFacade.getUsername()
 
-    // check if client is allowed to poll
-    val dataConsumer = dataConsumerRepository.findById(oauthClient)
-      ?: throw RuntimeException("Client $oauthClient is not a known data consumer")
-
-    val allowedTypes = dataConsumer.allowedEventTypes.split(",").map {
-      EventType.valueOf(it)
-    }
-
-    // checks types requested are allowed
-    val eventTypesToPoll = getTypeToPoll(eventTypes, allowedTypes, dataConsumer.clientName)
-
-    // Retrieve events from event stream
     val now = LocalDateTime.now()
-    val beginTime = fromTime ?: dataConsumer.lastPollEventTime ?: now.minusDays(1)
     val lastTime = toTime ?: now
-    val events = eventDataRepository.findAllByEventTypes(
-      eventTypesToPoll.map { it.toString() },
-      beginTime,
-      lastTime
-    ).map {
-      SubscribedEvent(
-        eventType = EventType.valueOf(it.eventType),
-        eventId = it.eventId,
-        eventTime = it.whenCreated!!
-      )
-    }
 
-    // save the time of the last record
-    val lastPollEventTime = events.lastOrNull()?.eventTime ?: lastTime
-    if (dataConsumer.lastPollEventTime == null || lastPollEventTime.isAfter(dataConsumer.lastPollEventTime)) {
-      dataConsumerRepository.save(dataConsumer.copy(lastPollEventTime = lastPollEventTime))
-    }
-
-    log.info("Client: {}: Retrieved {} events between {} and {} for event types {}", dataConsumer.clientName, events.count(), beginTime, lastTime, eventTypesToPoll)
-    return events
+    return consumerSubscriptionRepository.findAllByPollerClientId(authenticationFacade.getUsername())
+      .filter { eventTypes.isNullOrEmpty() || it.eventTypeId in eventTypes }
```

Comment:
So are we saying no event types provided means you get all? I think that may be wrong.

Reply:
No you just get the types you have subscribed to